internal/delta_map_downloader.go

```diff
@@ -40,9 +41,17 @@ func getDeltaRange(firstUsedLsn, firstNotUsedLsn uint64) (DeltaNo, DeltaNo) {
 	return firstUsedDeltaNo, firstNotUsedDeltaNo
 }
 
-func getWalSegmentRange(firstNotUsedDeltaNo DeltaNo, firstNotUsedLsn uint64) (WalSegmentNo, WalSegmentNo) {
-	firstUsedWalSegmentNo := firstNotUsedDeltaNo.firstWalSegmentNo()
-	lastUsedLsn := firstNotUsedLsn - 1
-	lastUsedWalSegmentNo := newWalSegmentNo(lastUsedLsn)
-	return firstUsedWalSegmentNo, lastUsedWalSegmentNo.next()
```

Comment:
This logic was correct even when there was switch of wall segments. Because in this case, segment which contained backup start LSN was lastUsedWalSegmentNo.next()

Reply:
`lastUsedLsn := firstNotUsedLsn - 1` points to the same segment as firstNotUsedLsn due to some WAL record right before redo point
![image](https://user-images.githubusercontent.com/47750602/89217415-9909ba00-d5e5-11ea-98aa-cb484317e935.png)

Seems like this record is being created every time on wal segment switch during pg_start_backup()
**TODO** find the source code to find the place where this record is being created